confluent_kafka/__init__.py

```diff
@@ -13,7 +16,633 @@
+
+        if not f.set_running_or_notify_cancel():
+            raise RuntimeError("Future was cancelled prematurely")
+
+        return f, futmap
+
+    def create_topics(self, new_topics, **kwargs):
+        """
+        Create new topics in cluster.
+
+        The future result() value is None.
+
+        :param new_topics list(NewTopic): New topics to be created.
+        :param operation_timeout float: Set broker's operation timeout in seconds,
+                  controlling how long the CreateTopics request will block
+                  on the broker waiting for the topic creation to propagate
+                  in the cluster. A value of 0 returns immediately. Default: 0
+        :param request_timeout float: Set the overall request timeout in seconds,
+                  including broker lookup, request transmission, operation time
+                  on broker, and response. Default: `socket.timeout.ms*1000.0`
+        :param validate_only bool: Tell broker to only validate the request,
+                  without creating the topic. Default: False
+
+        :returns: a dict of futures for each topic, keyed by the topic name.
+        :rtype dict(<topic_name, future>):
+
+        :raises KafkaException: Operation failed locally or on broker.
+        :raises TypeException: Invalid input.
+        :raises ValueException: Invalid input.
+        """
+
+        f, futmap = AdminClient._make_futures([x.topic for x in new_topics],
+                                              None,
+                                              AdminClient._make_topics_result)
+
+        super(AdminClient, self).create_topics(new_topics, f, **kwargs)
+
+        return futmap
+
+    def delete_topics(self, topics, **kwargs):
+        """
+        Delete topics.
+
+        The future result() value is None.
+
+        :param topics list(str): Topics to mark for deletion.
+        :param operation_timeout float: Set broker's operation timeout in seconds,
+                  controlling how long the DeleteTopics request will block
+                  on the broker waiting for the topic deletion to propagate
+                  in the cluster. A value of 0 returns immediately. Default: 0
+        :param request_timeout float: Set the overall request timeout in seconds,
+                  including broker lookup, request transmission, operation time
+                  on broker, and response. Default: `socket.timeout.ms*1000.0`
+
+        :returns: a dict of futures for each topic, keyed by the topic name.
+        :rtype dict(<topic_name, future>):
+
+        :raises KafkaException: Operation failed locally or on broker.
+        :raises TypeException: Invalid input.
+        :raises ValueException: Invalid input.
+        """
+
+        f, futmap = AdminClient._make_futures(topics, None,
+                                              AdminClient._make_topics_result)
+
+        super(AdminClient, self).delete_topics(topics, f, **kwargs)
```

Comment:
overriding methods like this seems confusing, better to use composition.

Reply:
I'm not an OO guy, as you are very well aware of. So please educate me what the practical difference is between composition and inheritance in this case?